pkg/dockerregistry/server/auditblobstore.go

```diff
@@ -0,0 +1,67 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+
+	"github.com/openshift/origin/pkg/dockerregistry/server/audit"
+)
+
+// auditBlobStore wraps a distribution.BlobStore to track operation result and
+// write it in the audit log.
+type auditBlobStore struct {
+	store distribution.BlobStore
+}
+
+var _ distribution.BlobStore = &auditBlobStore{}
+
+func (b *auditBlobStore) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
+	desc, err := b.store.Stat(ctx, dgst)
+	audit.GetLogger(ctx).Print(err, "BlobStore.Stat")
+	return desc, err
+}
+
+func (b *auditBlobStore) Get(ctx context.Context, dgst digest.Digest) ([]byte, error) {
+	blob, err := b.store.Get(ctx, dgst)
+	audit.GetLogger(ctx).Print(err, "BlobStore.Get")
```

Comment:
@smarterclayton Yes (according to log level). And all messages from lower levels will also contain your username and user ID. But this wrapper is made to ensure that the server log level will not affect audit log. Also I have a plan to use it for collecting statistics in the future.